spec/requests/omniauth_callbacks_controller_spec.rb

```diff
@@ -2,20 +2,24 @@
 
 require 'spec_helper'
 
-describe OmniauthCallbacksController, type: :request do
+# Devise calls OmniauthCallbacksController for OpenID Connect callbacks.
+describe '/user/spree_user/auth/openid_connect/callback', type: :request do
   include AuthenticationHelper
 
-  OmniAuth.config.test_mode = true
+  let(:user) { create(:user) }
+  let(:params) { { code: 'code123' } }
 
-  subject do
+  before do
+    OmniAuth.config.test_mode = true
     login_as user
-    post '/user/spree_user/auth/openid_connect/callback', params: { code: 'code123' }
   end
 
-  let(:user) { create(:user) }
+  def request!
```

Comment:
you could also use named subject ie :  `subject(:request!) { post(self.class.top_level_description, params:) } `

Reply:
Yes, I could. But then you can't repeat the request within the same `it` block because it's memoised. I think that it's better to distinguish between an action (method) and data (variable/let/subject).